server/tests/api.test.js

```diff
@@ -43,65 +55,79 @@ describe('API', () => {
-            const response = await request(API.app).get('/api/games');
-            const responseBody = JSON.parse(response.text);
-            const returnedKeys = Object.keys(responseBody);
-            expect(returnedKeys.length).toBe(1);
-
-            const returnedObject = responseBody[returnedKeys];
-            expect(returnedObject).toEqual(incorrectQueryParamsError);
-        });
-        xit('/api/game', async () => {
-        });
-        xit('/api/winrate', async () => {
-        });
-        xit('/api/improvement', async () => {
-        });
-        xit('/api/improvement-rate', async () => {
-        });
+        [
+            "games",
+            "game",
+            "winrate",
+            "improvement",
+            "improvement-rate",
+        ].forEach(route => {
+            it(`/api/${route}`, async () => {
+                const response = await request(API.app).get(`/api/${route}`);
+                const responseBody = JSON.parse(response.text);
+                const returnedKeys = Object.keys(responseBody);
+                expect(returnedKeys.length).toBe(1);
+    
+                const returnedObject = responseBody[returnedKeys];
+                expect(returnedObject).toEqual(incorrectQueryParamsError);
+            });
+        })
     });
 
     // TODO: Test it doesn't crash
+    
     describe('Queries With Badly Typed Arguments', () => {
-        it('/api/games', async () => {
-            const response = await request(API.app)
-                .get('/api/games')
-                .query({ page: 'page 1' });
-            const responseBody = JSON.parse(response.text);
-            const returnedKeys = Object.keys(responseBody);
-            expect(returnedKeys.length).toBe(1);
-
-            const returnedObject = responseBody[returnedKeys];
-            expect(returnedObject).toEqual(incorrectQueryParamsError);
-        });
-        xit('/api/game', async () => {
-        });
-        xit('/api/winrate', async () => {
-        });
-        xit('/api/improvement', async () => {
-        });
-        xit('/api/improvement-rate', async () => {
-        });
+        Object.entries({
+            "games": { page: 'page 1' },
+            "game": { gameId: "stillnot36" },
+            "winrate": { agentId: "not36again" },
+            "improvement": { agentId: "not36" },
+            "improvement-rate": { agentId: "" },
+        }).forEach(([endpoint, query]) => {
+            it(`/api/${endpoint}`, async () => {
+                const response = await request(API.app)
+                    .get(`/api/${endpoint}`)
+                    .query(query);
+                const responseBody = JSON.parse(response.text);
+                const returnedKeys = Object.keys(responseBody);
+                expect(returnedKeys.length).toBe(1);
+                const returnedObject = responseBody[returnedKeys];
+                expect(returnedObject).toEqual(incorrectQueryParamsError);
+            });
+        })
```

Comment:
The reason I did this all separately is actually so only the broken API endpoints would fail a test
We can write a function that takes `endpoint` and `query` and runs this block.

You've done so much good work though, I can take this off your hands if you'd like.

Reply:
good idea, i didn't think of that, that would be great